### PR TITLE
More nightly clippy fixes

### PR DESCRIPTION
**Pull request by Manishearth**
A lot of this is reformatting docs so that there is a clear, short summary line.

### CODE REVIEW

File: components/collator/src/elements.rs [hunk 1]
```diff
@@ -116,13 +116,12 @@ const COMMON_SEC_AND_TER_CE: u64 = COMMON_SECONDARY_CE | COMMON_TERTIARY_CE;
 
 const UNASSIGNED_IMPLICIT_BYTE: u8 = 0xFE;
 
-/// Set if there is no match for the single (no-suffix) character itself.
-/// This is only possible if there is a prefix.
-/// In this case, discontiguous contraction matching cannot add combining marks
-/// starting from an empty suffix.
-/// The default CE32 is used anyway if there is no suffix match.
+// /// Set if there is no match for the single (no-suffix) character itself.
```
Comment (robertbastian) [inline components/collator/src/elements.rs:119]: issue

File: components/collator/tests/tests.rs [hunk 2]
```diff
@@ -1698,7 +1698,7 @@ fn test_conformance_shifted() {
             if line.is_empty() {
                 continue;
             }
-            if line.starts_with(&[b'#']) {
+            if line.starts_with(b"#") {
```
Comment (robertbastian) [inline components/collator/tests/tests.rs:1701]: this can be `b'#'`?

File: components/collator/src/elements.rs (hunk 1 above)
Comment (Manishearth) [reply components/collator/src/elements.rs:119]: It's correct, but there should be an extra newline making it clear

File: components/collator/tests/tests.rs [hunk 3]
```diff
@@ -1741,7 +1741,7 @@ fn test_conformance_non_ignorable() {
...
```
Comment (robertbastian) [inline components/collator/tests/tests.rs:1744]: here too

Comment (Manishearth) [reply components/collator/tests/tests.rs:1744]: Nope: it doesn't work. `starts_with` only accepts Patterns in the case of strings, not slices

Comment (robertbastian) [reply components/collator/tests/tests.rs:1744]: but it used a pattern beofre?

Comment (Manishearth) [reply components/collator/tests/tests.rs:1744]: No, it used a slice.

Comment (Manishearth) [reply components/collator/tests/tests.rs:1744]: slices happen to be Patterns, however this function accepts slices only

Comment (robertbastian) [reply components/collator/tests/tests.rs:1744]: not sure this is an improvement

**Review by robertbastian — APPROVED** [review]

**Pull request merged**